test/input_gen/recorder.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: Apache-2.0
+##
+# Copyright (C) 2020 Jihoon Lee <jhoon.it.lee@samsung.com>
+#
+# @file recorder.py
+# @brief Generate tc from given keras model
+# @author Jihoon lee <jhoon.it.lee@samsung.com>
+
+from functools import wraps
+import sys
+import os
+import warnings
+import random
+
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore", category=FutureWarning)
+    import numpy as np
+    import tensorflow as tf
+    from tensorflow.python import keras as K
+
+tf.compat.v1.enable_eager_execution()
+# Fix the seeds across frameworks
+SEED = 1234
+random.seed(SEED)
+tf.compat.v1.set_random_seed(SEED)
+np.random.seed(SEED)
+
+
+##
+# Keras Recorder
+
+##
+# @brief Record Keras model with some watchers attached
+# @note  The class might need to go through some rework for non-sequential model
+# in case of the order of graph traversal is diffrent from NNTrainer
+class KerasRecorder:
+    def __init__(
+        self,
+        file_name,
+        inputs,
+        outputs,
+        input_shape,
+        label_shape,
+        loss_fn=None,
+        optimizer=tf.keras.optimizers.SGD(lr=1.0),
```

Comment:
Maybe in a later PR, but you would want another argument here `from_logits`.
In this case, the activation for the last layer (for eg softmax) is amalgamated in the loss itself.
So, the output of the last layer must add that activation by itself (just for returning and not for training) to match the output with nntrainer.
This is needed because amalgamated losses are mathematically more stable than the separated versions.

Reply:
Isn't this done by adding 

`loss_fn=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)`
and omit softmax activation on he last layer?

Reply:
Yes, this is done for training, but inference, softmax is still needed (which requires extra conditions than just setting `from_logits=True`).

Reply:
Ditto. I will do this in diffrent PR. For now, I'll leave a note in my issue